jerry-core/parser/js/js-parser.c

```diff
@@ -2707,10 +2706,18 @@ parser_parse_arrow_function (parser_context_t *context_p, /**< context */
   parser_saved_context_t saved_context;
   ecma_compiled_code_t *compiled_code_p;
 
+#ifndef CONFIG_DISABLE_ES2015_CLASS
+  uint32_t class_has_super = context_p->status_flags & PARSER_CLASS_HAS_SUPER;
+#endif /* !CONFIG_DISABLE_ES2015_CLASS */
+
   JERRY_ASSERT ((status_flags & PARSER_IS_FUNCTION)
                  && (status_flags & PARSER_IS_ARROW_FUNCTION));
   parser_save_context (context_p, &saved_context);
+#ifndef CONFIG_DISABLE_ES2015_CLASS
+  context_p->status_flags |= status_flags | PARSER_ARGUMENTS_NOT_NEEDED | class_has_super;
```

Comment:
Remove the `uint32_t class_has_super = context_p->status_flags & PARSER_CLASS_HAS_SUPER;` above.

Add `context_p->status_flags |= context_p->status_flags & PARSER_CLASS_HAS_SUPER` here if the line above is too long with this.

Reply:
If you save the context you lose all of your status flags except the `PARSER_IS_STRICT`, so 
`context_p->status_flags |= context_p->status_flags & PARSER_CLASS_HAS_SUPER` will never add the `PARSER_CLASS_HAS_SUPER` flag to the new context.

But we can combine our solutions like:
```
context_p->status_flags |= saved_context.status_flags & PARSER_CLASS_HAS_SUPER
```